NOAA-FIMS/FIMS
# Unit test 180 feat population

Thanks for opening a pull request (PR) to FIMS! Please describe the changes under the headings below. Please also go through the checklist below to ensure the appropriate steps have been taken during development. For smaller changes, feel free to skip steps that aren't relevant, but at least describe the feature under the "what is the feature".

# What is the feature?
Merge changes from unit test branch to population branch and see if we can resolve some merge conflicts before writing more C++ unit tests.

# How have you implemented the solution?
Add a few C++ unit tests and also updated population.hpp

# Does it impact any other area of the project?
May create merge conflicts when merging other branches to the population branch. 

# Does this change impact the model input or output?*
No

# How to test this change
The C++ tests can be found under tests/gtest. run-googletest and call-r-cmd-check GHA workflows did not run successfully and we will fix the tests on Dec 9, 2022.

# Developer pre-PR Checklist

Please do these steps locally for big changes. Note GitHub Actions does all of these checks automatically when pushing to the repository. Please see [code development section in the contributor guide](https://noaa-fims.github.io/collaborative_workflow/contributor-guidelines.html#code-development) for details.
